boa_interner/src/raw.rs

```diff
@@ -0,0 +1,196 @@
+use std::hash::Hash;
+
+use rustc_hash::FxHashMap;
+
+use crate::{fixed_string::FixedString, interned_str::InternedStr};
+
+/// Raw string interner, generic by a char type.
+#[derive(Debug)]
+pub(super) struct RawInterner<Char> {
+    // COMMENT FOR DEVS:
+    // This interner works on the assumption that
+    // `head` won't ever be reallocated, since this could invalidate
+    // some of our stored pointers inside `spans`.
+    // This means that any operation on `head` and `full` should be carefully
+    // reviewed to not cause Undefined Behaviour.
+    // `get_or_intern` has a more thorough explanation on this.
+    //
+    // Also, if you want to implement `shrink_to_fit` (and friends),
+    // please check out https://github.com/Robbepop/string-interner/pull/47 first.
+    // This doesn't implement that method, since implementing it increases
+    // our memory footprint.
+    symbol_cache: FxHashMap<InternedStr<Char>, usize>,
+    spans: Vec<InternedStr<Char>>,
+    head: FixedString<Char>,
+    full: Vec<FixedString<Char>>,
+}
+
+impl<Char> Default for RawInterner<Char> {
+    fn default() -> Self {
+        Self {
+            symbol_cache: FxHashMap::default(),
+            spans: Vec::default(),
+            head: FixedString::default(),
+            full: Vec::default(),
+        }
+    }
+}
+
+impl<Char> RawInterner<Char> {
+    /// Creates a new [`Interner`] with the specified capacity.
```

Comment:
Need to update the comments to the correct struct name, here and a few more comments
```suggestion
    /// Creates a new [`RawInterner`] with the specified capacity.
```